### PR TITLE
Bump python support up to 3.11

### PR DESCRIPTION
**Pull request by tmwly**
This PR adds support for python 3.8, 3.9, 3.10 and 3.11, and bumps the plugin code env dependencies where necessary

Also, we update the plugin desc a bit 

[sc-132114]

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2021 Dataiku
+   Copyright 2021 - 2023 Dataiku SAS
```
Comment (AgatheG) [inline LICENSE:189]: Should probably be simply 2023 here?

File: plugin.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
   "id" : "model-error-analysis",
-  "version" : "1.2.0",
+  "version" : "1.2.1",
   "meta" : {
     "label" : "Model Error Analysis",
-    "description" : "Debug model performance with error analysis",
+    "description" : "Debug model performance with error analysis. A code env is only required to use the example Jupyter Notebook.",
```
Comment (AgatheG) [inline plugin.json:6]:
```suggestion
    "description" : "Debug model performance with error analysis. A code env is only required to use the Jupyter notebook.",
```

**Comment by AgatheG** [comment]
I could not create the plugin code env for py3.8+ (only managed to have it for 3.6 and 3.7) 😕

File: plugin.json (hunk 2 above)
Comment (AgatheG) [reply plugin.json:6]: you can keep the upper cvase then! I'll simply remove "example"

**Comment by tmwly** [comment]
Issue was related to using a specific wheel in certain cases with scipy-1.2.3, which would succeed, but would fail if the wheel was unavailable. Fixed by bumping scipy to a higher version for 3.8+

File: plugin.json (hunk 2 above)
Comment (tmwly) [reply plugin.json:6]: perfect !

**Review by AgatheG — APPROVED** [review]
:chefkiss: LGTM!

**Pull request merged**